File: rest/client/src/app/models/dtos/contribution/responses/get-contribution-overview-dto/get-contribution-overview-dto.ts
export class GetContributionOverviewDto {
    constructor(
        public contributionID: string,
        public projectID: string,
        public userID: string,
        public category: string,
        public title: string,
        public text: string,
        public date: string,
        public status: string,
        public votes: VoteDto[],
        public comments: CommentDto[],
      ) {}
}

export class VoteDto {
  constructor(
    public voteID: string,
    public userID: string,
  ) {}
}

export class CommentDto {
  constructor(
    public commentID: string,
  ) {}
}

File: rest/server/Server/Models/DTOs/Contribution/Responses/GetContributionOverviewDto.cs
namespace Server.Models.DTOs.Contribution.Responses
{
    public class GetContributionOverviewDto
    {
        public Guid ContributionID { get; set; }
        public Guid ProjectID { get; set; }
        public Guid UserID { get; set; }
        public string Category { get; set; }
        public string Title { get; set; }
        public string Text { get; set; }
        public string Date { get; set; }
        public string Status { get; set; }
        public List<VoteDto> Votes { get; set; } = new();
        public List<CommentDto> Comments { get; set; } = new();
    }

    public class VoteDto
    {
        public Guid VoteID { get; set; }
        public Guid UserID { get; set; }
    }

    public class CommentDto
    {
        public Guid CommentID { get; set; }
    }
}
